portal-kernel/src/com/liferay/portal/kernel/util/StringUtil.java

```diff
@@ -4671,13 +4671,17 @@ public static String upperCase(String s) {
 	 * @return the string, with its first character converted to upper-case
 	 */
 	public static String upperCaseFirstLetter(String s) {
-		char[] chars = s.toCharArray();
+		if (!s.isEmpty()) {
```

Comment:
Hi Lazlo,

I would also add null check. If the method parameter is null then return null instead of throwing NPE.

BTW, good optimization.